src/ert/run_models/everest_run_model.py

```diff
@@ -420,44 +420,89 @@ def _on_before_forward_model_evaluation(
             logging.getLogger(EVEREST).info("User abort requested.")
             optimizer.abort_optimization()
 
+    @staticmethod
+    def _consolidate_cache_and_sim_results(
+        num_simulations: int,
+        cached_results: dict[int, Any],
+        simulated_results: NDArray[np.float64] | None = None,
+    ) -> NDArray[np.float64] | None:
+        if simulated_results is None:
+            return np.array(
+                [cached_results.get(sim) for sim in range(num_simulations)],
+                dtype=np.float64,
+            )
```

Comment:
This assumes that `cached_results` contains all results and they are the expected results. I do not think that is garantueed, see my comments below at the point where the cached values are retrieved.  If not all values are present, this will fail, if wrong values are present they will end up in the results.